src/info/mod.rs

```diff
@@ -364,4 +371,52 @@ mod tests {
         );
         Ok(())
     }
+
+    type Result<T = ()> = std::result::Result<T, Box<dyn std::error::Error>>;
+
+    fn test_repo(name: &str) -> Result<Repository> {
+        let repo_path = git_testtools::scripted_fixture_repo_read_only(name)?;
+        let safe_repo = ThreadSafeRepository::open_opts(repo_path, open::Options::isolated())?;
+        Ok(safe_repo.to_thread_local())
+    }
+
+    #[test]
+    fn test_bare_repo() -> Result {
+        let repo = test_repo(&"bare_repo.sh")?;
+        match Info::init_repo_path(&repo) {
+            Ok(_info) => assert!(false, "oops, info was returned on a bare git repo"),
+            Err(error) => assert_eq!(
+                "please run onefetch inside of a non-bare git repository",
+                error.to_string()
+            ),
+        };
```

Comment:
Another nitpick: instead of asserting false, maybe assert `.is_err()`, then unwrap the error for the following `assert_eq!`.

Reply:
Sweet, so much  better.
```rust
        let res = Info::init_repo_path(&repo);
        assert!(res.is_err(), "oops, info was returned on a bare git repo");
        assert_eq!(
            res.unwrap_err().to_string(),
            "please run onefetch inside of a non-bare git repository"
        );
```